### DISCUSSION TITLE
Possible to create OpenTK bindings?

### DISCUSSION THREAD

**Discussion by mludlum** [StereoKit/StereoKit#1056] [Q&A] [answered]
I'm experimenting with development on my Quest 3.  
I have 3D codebase that uses OpenTK for 3D visualization. The code runs on Windows and Android. Is it possible to create bindings for OpenGL including compiling and loading shaders?   Would using GetFunction/Ptr be a way to do this?

**Comment by maluoi** [comment, marked as answer]
If you're using StereoKit, it's not exactly easy to manually mix in direct rendering code! StereoKit is a whole renderer pipeline of its own, and manages the graphics assets, render surfaces and timing at a pretty low level compared to the public API. Not to mention SK uses different graphics APIs on Windows vs Android. If you are porting an app, I'd recommend just using StereoKit's rendering code and shaders instead.

If there are still items you cannot do and you don't need to render directly to the display surface, direct calls to GL can be okay. While you could write your own bindings, I'd probably recommend something like the existing [Silk bindings](https://github.com/dotnet/Silk.NET) to OpenGL that should simplify the process a fair bit.
